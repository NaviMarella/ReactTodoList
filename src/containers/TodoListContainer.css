.todoList__header{
    display: flex;
    justify-content: space-between;
    flex-direction: row;
    width: 100%;
    padding: 20px 0;
}

.todoList__header div{
    font-size: 20px;
    font-weight: bold;
    color: gray;
}

.todolist__count{
    justify-content: flex-end;
}