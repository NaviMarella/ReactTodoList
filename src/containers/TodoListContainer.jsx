import React from "react";
import PropTypes from 'prop-types';
import Box from "../components/Box";
import Label from "../components/Label";
import TodoListItem from "../components/TodoListItem";
import "./TodoListContainer.css";
import Conditional from '../components/Conditional';

const TodoListContainer = ({todoList, dispatch, currentLabel}) => {
    let todos = todoList.filter(todo => todo && !todo.isCompleted);
    return(
         <Conditional condition={todoList.length > 0}>
             <Box type="large">
                <div className="todoList__header">
                    <Label text = {`Todo Items:   ${currentLabel}`} />
                    <Label className="todolist__count" text = {`${todos.length}/${todoList.length}`}/>
                </div>
                {
                    todoList.map(item => {
                        return (
                            <TodoListItem key={item.id} todo={item} currentLabel={currentLabel} dispatch={dispatch}/>
                        )
                    })
                }
            </Box>
         </Conditional>
    )
}

TodoListContainer.defaultProps = {
    todoList: []
}

TodoListContainer.propTypes = {
    todoList: PropTypes.array.isRequired,
    currentLabel: PropTypes.string,
    dispatch: PropTypes.func.isRequired
};

export default TodoListContainer