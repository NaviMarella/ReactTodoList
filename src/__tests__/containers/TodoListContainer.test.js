import { shallow } from 'enzyme';
import React from 'react';
import TodoListContainer from './../../containers/TodoListContainer';

describe('TodoListContainer Component', () => {
  test('render snapshot', () => {
    let props = {
        todoList: [
            {
                id: 1,
                label: 'TODOS',
                text: "Learn React",
                isCompleted: true
            }
        ],
        currentLabel: "TODOS",
        dispatch: jest.fn()
    }
    let wrapper = shallow(<TodoListContainer {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
