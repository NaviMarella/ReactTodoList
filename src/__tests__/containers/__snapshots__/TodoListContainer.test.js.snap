// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListContainer Component render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoListContainer
    currentLabel="TODOS"
    dispatch={[MockFunction]}
    todoList={
      Array [
        Object {
          "id": 1,
          "isCompleted": true,
          "label": "TODOS",
          "text": "Learn React",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Box
        type="large"
      >
        <div
          className="todoList__header"
        >
          <Label
            text="Todo Items:   TODOS"
          />
          <Label
            className="todolist__count"
            text="0/1"
          />
        </div>
        <TodoListItem
          currentLabel="TODOS"
          dispatch={[MockFunction]}
          todo={
            Object {
              "id": 1,
              "isCompleted": true,
              "label": "TODOS",
              "text": "Learn React",
            }
          }
        />
      </Box>,
      "condition": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="todoList__header"
          >
            <Label
              text="Todo Items:   TODOS"
            />
            <Label
              className="todolist__count"
              text="0/1"
            />
          </div>,
          Array [
            <TodoListItem
              currentLabel="TODOS"
              dispatch={[MockFunction]}
              todo={
                Object {
                  "id": 1,
                  "isCompleted": true,
                  "label": "TODOS",
                  "text": "Learn React",
                }
              }
            />,
          ],
        ],
        "type": "large",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Label
                text="Todo Items:   TODOS"
              />,
              <Label
                className="todolist__count"
                text="0/1"
              />,
            ],
            "className": "todoList__header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "Todo Items:   TODOS",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "todolist__count",
                "text": "0/1",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "currentLabel": "TODOS",
            "dispatch": [MockFunction],
            "todo": Object {
              "id": 1,
              "isCompleted": true,
              "label": "TODOS",
              "text": "Learn React",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Box
          type="large"
        >
          <div
            className="todoList__header"
          >
            <Label
              text="Todo Items:   TODOS"
            />
            <Label
              className="todolist__count"
              text="0/1"
            />
          </div>
          <TodoListItem
            currentLabel="TODOS"
            dispatch={[MockFunction]}
            todo={
              Object {
                "id": 1,
                "isCompleted": true,
                "label": "TODOS",
                "text": "Learn React",
              }
            }
          />
        </Box>,
        "condition": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="todoList__header"
            >
              <Label
                text="Todo Items:   TODOS"
              />
              <Label
                className="todolist__count"
                text="0/1"
              />
            </div>,
            Array [
              <TodoListItem
                currentLabel="TODOS"
                dispatch={[MockFunction]}
                todo={
                  Object {
                    "id": 1,
                    "isCompleted": true,
                    "label": "TODOS",
                    "text": "Learn React",
                  }
                }
              />,
            ],
          ],
          "type": "large",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Label
                  text="Todo Items:   TODOS"
                />,
                <Label
                  className="todolist__count"
                  text="0/1"
                />,
              ],
              "className": "todoList__header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "text": "Todo Items:   TODOS",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "todolist__count",
                  "text": "0/1",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "currentLabel": "TODOS",
              "dispatch": [MockFunction],
              "todo": Object {
                "id": 1,
                "isCompleted": true,
                "label": "TODOS",
                "text": "Learn React",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
